tools/monorepo-utils/src/md-docs/lib/generate-frontmatter.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import matter from 'gray-matter';
+import yaml from 'js-yaml';
+
+/**
+ * Generate front-matter for supported post attributes.
+ *
+ * @param fileContents
+ */
+export const generatePostFrontMatter = (
+	fileContents: string
+): {
+	[ key: string ]: unknown;
+} => {
+	const allowList = [
+		'post_date',
+		'post_title',
+		'page_template',
+		'post_author',
+		'post_name',
+		'category_title',
+		'category_slug',
+	];
+
+	const frontMatter = matter( fileContents, {
+		engines: {
+			// Disable date parsing so that we can get the raw date string.
```

Comment:
Which option disables the date parsing here? Not obvious to me. 🤔

Reply:
Ah yeah its passing JSON_SCHEMA to the yaml parser so it treats dates as they are treated in json. Ill add a link to the suggested workaround that discusses this in that repo.

Reply:
Addressed [here](https://github.com/woocommerce/woocommerce/pull/39214/commits/e06fff0903935df14b39d64eb070fe2c35d0ccba)